tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/workspace/LiferayWorkspaceImportOpMethods.java

```diff
@@ -0,0 +1,116 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.workspace;
+
+import com.liferay.ide.core.LiferayCore;
+import com.liferay.ide.project.core.NewLiferayProjectProvider;
+import com.liferay.ide.project.core.ProjectCore;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.sapphire.modeling.ProgressMonitor;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.platform.ProgressMonitorBridge;
+import org.eclipse.wst.server.core.IRuntimeWorkingCopy;
+import org.eclipse.wst.server.core.IServerWorkingCopy;
+import org.eclipse.wst.server.core.ServerCore;
+
+/**
+ * @author Andy Wu
+ */
+public class LiferayWorkspaceImportOpMethods
+{
+
+    public static final Status execute( final LiferayWorkspaceImportOp op, final ProgressMonitor pm )
+    {
+        final IProgressMonitor monitor = ProgressMonitorBridge.create( pm );
+
+        monitor.beginTask( "Importing Liferay Workspace project...", 100 );
+
+        Status retval = Status.createOkStatus();
+
+        try
+        {
+            @SuppressWarnings( "unchecked" )
+            NewLiferayProjectProvider<NewLiferayWorkspaceOp> provider =
+                (NewLiferayProjectProvider<NewLiferayWorkspaceOp>) LiferayCore.getProvider( "liferay-workspace" );
+
+            String location = op.getWorkspaceLocation().content().toOSString();
+
+            LiferayWorkspaceUtil.clearWorkspace( location );
+
+            boolean isInitBundle = op.getRunInitBundleCommand().content();
+            boolean isHasBundlesDir = op.getHasBundlesDir().content();
+
+            if( isInitBundle && !isHasBundlesDir )
+            {
+                provider.importProject( location, monitor, "initBundle" );
+            }
+            else
+            {
+                provider.importProject( location, monitor, null );
+            }
+            boolean isAddServer = op.getAddServer().content();
+
+            if( isAddServer )
+            {
+                addPortalRuntimeAndServer( location, monitor );
+            }
+        }
+        catch( Exception e )
+        {
+            final String msg = "import Liferay Workspace project error";
+
+            ProjectCore.logError( msg, e );
+
+            retval = Status.createErrorStatus( msg, e );
+        }
+
+        return retval;
+    }
+
+    private static void addPortalRuntimeAndServer( String location, IProgressMonitor monitor ) throws CoreException
+    {
+        SimpleDateFormat formator = new SimpleDateFormat( "yyyyMMddHHmmss" );
```

Comment:
Lets make the name of the server configurable by the user in the wizard and the default name of the server property will be "workspace directory name" server.